packages/swi_prolog.rb

```diff
@@ -0,0 +1,49 @@
+require 'package'
+require 'fileutils'
+
+class Swi_prolog < Package
+  description 'SWI-Prolog offers a comprehensive free Prolog environment. Since its start in 1987, SWI-Prolog development has been driven by the needs of real world applications. SWI-Prolog is widely used in research and education as well as commercial applications. Join over a million users who have downloaded SWI-Prolog.'
+  homepage 'https://www.swi-prolog.org/'
+  version '8.2.4'
+  compatibility 'all'
+  source_url 'https://www.swi-prolog.org/download/stable/src/swipl-8.2.4.tar.gz'
+  source_sha256 'f4bcc78437f9080ab089762e9e6afa7071df7f584c14999b92b9a90a4efbd7d8'
+
+  binary_url ({
+  })
+  binary_sha256 ({
+  })
```

Comment:
Missing binaries?

Reply:
Seeing a compile failure on armv7l:
```
[106/106] Generating ../home/boot.prc
FAILED: home/boot.prc
cd /usr/local/tmp/crew/swipl-8.2.4.tar.gz.dir/swipl-8.2.4/builddir/src && /usr/local/bin/cmake -E remove -f /usr/local/tmp/crew/swipl-8.2.4.tar.gz.dir/swipl-8.2.4/builddir/home/boot.prc && /usr/local/tmp/crew/swipl-8.2.4.tar.gz.dir/swipl-8.2.4/builddir/src/swipl -q -O -o /usr/local/tmp/crew/swipl-8.2.4.tar.gz.dir/swipl-8.2.4/builddir/home/boot.prc -b /usr/local/tmp/crew/swipl-8.2.4.tar.gz.dir/swipl-8.2.4/builddir/home/boot/init.pl
[FATAL ERROR: at Tue Feb  9 18:00:49 2021
        WAM Table mismatch: wam_table[8(h_mpz)] == wam_table[10(h_string)]
]
ninja: build stopped: subcommand failed.
PGO: run ninja to complete the build
[1/499] Generating VMI metadata
[2/499] Generating static atoms and functors
[3/418] Checking (git) version
[4/418] Generating ../home/boot.prc
FAILED: home/boot.prc
cd /usr/local/tmp/crew/swipl-8.2.4.tar.gz.dir/swipl-8.2.4/builddir/src && /usr/local/bin/cmake -E remove -f /usr/local/tmp/crew/swipl-8.2.4.tar.gz.dir/swipl-8.2.4/builddir/home/boot.prc && /usr/local/tmp/crew/swipl-8.2.4.tar.gz.dir/swipl-8.2.4/builddir/src/swipl -q -O -o /usr/local/tmp/crew/swipl-8.2.4.tar.gz.dir/swipl-8.2.4/builddir/home/boot.prc -b /usr/local/tmp/crew/swipl-8.2.4.tar.gz.dir/swipl-8.2.4/builddir/home/boot/init.pl
[FATAL ERROR: at Tue Feb  9 18:00:50 2021
        WAM Table mismatch: wam_table[8(h_mpz)] == wam_table[10(h_string)]
]
[5/418] Building C object packages/clib/CMakeFiles/plugin_files.dir/error.c.o
```

Reply:
Just excluding armv7l... this package moves fast enough that if someone wants to make a dev-package, it will probably compile fine on armv7l.

Reply:
Binaries added.